sparkler-api/src/main/java/edu/usc/irds/sparkler/util/FetcherDefault.java

```diff
@@ -39,6 +68,15 @@
     public FetchedData fetch(Resource resource) throws Exception {
         LOG.info("DEFAULT FETCHER {}", resource.getUrl());
         URLConnection urlConn = new URL(resource.getUrl()).openConnection();
+        if (httpHeaders != null){
+            httpHeaders.entrySet().forEach(e -> urlConn.setRequestProperty(e.getKey(), e.getValue()));
+        }
+        String userAgentValue = getUserAgent();
+        if (getUserAgent() != null) {
+            LOG.debug(USER_AGENT + ": " + userAgentValue);
+            urlConn.setRequestProperty(USER_AGENT, userAgentValue);
```

Comment:
(a) point taken. It has to be a list to rotate indices. I will support removal of duplicates using a different implementation of List.
(b) I had believed that YAML supports block comments. Now, I see it doesnt have block comments so I will move to a separate file as suggested